test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,114 @@
+package importcsv
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/konveyor/tackle2-hub/api"
+)
+
+func TestImportCSV(t *testing.T) {
+	for _, r := range TestCases {
+		t.Run(r.FileName, func(t *testing.T) {
+
+			// Upload CSV.
+			inputData := make(map[string]interface{})
+			err := Client.FilePost("/importsummaries/upload", r.FileName, &inputData)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Check list of Applications.
+			importedApps, _ := Application.List()
+			expectedApps := r.ExpectedApplications
+			for i, importedApp := range importedApps {
+				if i >= len(expectedApps) {
+					t.Errorf("Extra imported Application: %s", importedApp.Name)
+					continue
+				}
+				expectedApp := expectedApps[i].Name
+				if importedApp.Name != expectedApp {
+					t.Errorf("Mismatch in imported Application: Expected %s, Actual %s", expectedApp, importedApp.Name)
+				}
+			}
+
+			// Check list of Dependencies.
+			importedDeps, _ := Dependency.List()
+			expectedDeps := r.ExpectedDependencies
```

Comment:
Are these expectedDeps&expectedApps needed, wouldn't be better use r.Expected... variables?

Reply:
Will do that.